docs/database/mongodb-integration.md

```diff
@@ -1,137 +1,310 @@
+
+When the app host runs, the password is stored in the app host's secret store. It's added to the `Parameters` section, for example:
+
+```json
 {
-    // Use mongoClient...
+  "Parameters:mongo-password": "<THE_GENERATED_PASSWORD>"
 }
 ```
 
-After adding a `IMongoClient`, you can require the `IMongoClient` instance using DI.
+The name of the parameter is `mongo-password`, but really it's just formatting the resource name with a `-password` suffix. For more information, see [Safe storage of app secrets in development in ASP.NET Core](/aspnet/core/security/app-secrets) and [Add MongoDB server resource with parameters](#add-mongodb-server-resource-with-parameters).
+
+The <xref:Aspire.Hosting.ResourceBuilderExtensions.WithReference%2A> method configures a connection in the `ExampleProject` named `mongodb`.
+
+> [!TIP]
+> If you'd rather connect to an existing MongoDB server, call <xref:Aspire.Hosting.ParameterResourceBuilderExtensions.AddConnectionString*> instead. For more information, see [Reference existing resources](../fundamentals/app-host-overview.md#reference-existing-resources).
+
+### Add MongoDB server resource with data volume
+
+To add a data volume to the MongoDB server resource, call the <xref:Aspire.Hosting.MongoDBBuilderExtensions.WithDataVolume*> method on the MongoDB server resource:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
 
-## App host usage
+var mongo = builder.AddMongoDB("mongo")
+                   .WithDataVolume();
+
+var mongodb = mongo.AddDatabase("mongodb");
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(mongodb)
+       .WaitFor(mongodb);
+
+// After adding all resources, run the app...
+```
 
-To model the MongoDB resource in the app host, install the [📦 Aspire.Hosting.MongoDB](https://www.nuget.org/packages/Aspire.Hosting.MongoDB) NuGet package in the [app host](xref:dotnet/aspire/app-host) project.
+The data volume is used to persist the MongoDB server data outside the lifecycle of its container. The data volume is mounted at the `/data/db` path in the MongoDB server container and when a `name` parameter isn't provided, the name is generated at random. For more information on data volumes and details on why they're preferred over [bind mounts](#add-mongodb-server-resource-with-data-bind-mount), see [Docker docs: Volumes](https://docs.docker.com/engine/storage/volumes).
+
+> [!WARNING]
+> The password is stored in the data volume. When using a data volume and if the password changes, it will not work until you delete the volume.
+
+### Add MongoDB server resource with data bind mount
+
+To add a data bind mount to the MongoDB server resource, call the <xref:Aspire.Hosting.MongoDBBuilderExtensions.WithDataBindMount*> method:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var mongo = builder.AddMongoDB("mongo")
+                   .WithDataVolume();
```

Comment:
This is in the data bind mount section, but it is calling `.WithDataVolume`, I think it should only call WithDataBindMount.